fs/fs.go

```diff
@@ -149,12 +164,45 @@ func NewFilesystem(root string, cfg config.Config, opts ...Option) (_ snapshot.F
 		})
 	}
 
+	bgFetchPeriod := time.Duration(cfg.BackgroundFetchConfig.FetchPeriodMSec) * time.Millisecond
+	if bgFetchPeriod == 0 {
+		bgFetchPeriod = defaultBgFetchPeriod
+	}
+
+	bgSilencePeriod := time.Duration(cfg.BackgroundFetchConfig.SilencePeriodMSec) * time.Millisecond
+	if bgSilencePeriod == 0 {
+		bgSilencePeriod = defaultBgSilencePeriod
+	}
+
+	bgMaxQueueSize := cfg.BackgroundFetchConfig.MaxQueueSize
+	if bgMaxQueueSize == 0 {
+		bgMaxQueueSize = defaultBgMaxQueueSize
```

Comment:
How do users figure out the appropriate queue size? Do we have enough logs/metrics to tell users that the size is good?

Reply:
Making it configurable.